requirements_test_all.txt

```diff
@@ -6,6 +6,14 @@
 # homeassistant.components.aemet
 AEMET-OpenData==0.6.4
 
+# homeassistant.components.homeassistant_hardware
+# homeassistant.components.homeassistant_sky_connect
+# homeassistant.components.homeassistant_yellow
+# homeassistant.components.otbr
+# homeassistant.components.zha
+ha-silabs-firmware-client==0.2.0
+universal-silabs-flasher==0.0.30
```

Comment:
These are subdependencies, not direct ones?

Reply:
Aah I see. Let me ping @puddly in that case.

This change isn't correct though, as this file is generated and shouldn't thus be the only change (the sources needs to change).

Reply:
Oh, you're right. There's a banner on top of the file.

I'll try `script/gen_requirements_all.py` once my current test run is done. I wonder why wasn't it catched by CI though?

Reply:
Double checked, `gen_requirements_all.py` doesn't generate anything for silabs packages. Would you like me to close this PR and file a bug instead or @puddly will handle it?

Reply:
Yeah, I'm not sure why this is not working right. The hardware integration depends on these packages directly in its manifest and the Yellow and ZBT-1 integrations depend on the hardware one.

Reply:
Okay, I'm pretty sure it has something to do with dashes and underscores in import/package name (it's `universal_silabs_flasher` import but `universal-silabs-flasher` package). I have a very dirty fix to `gen_requirements_all.py` that confirms this.

Would you like to handle this, or I should prepare a PR?

Reply:
Okay, found and fixed the culprit. It was two-character fix. Ready for review.